### PR TITLE
Fix wrong field alias with query string search on deeply-inherited tables

### PR DESCRIPTION
**Pull request by fquffio**
This PR is basically an ugly workaround to alias fields the right way.

Visiting `GET /users?filter[query]=gustavo` now causes a 500 Internal Server Error because some fields are improperly aliased with `Profiles` instead of `Objects`. This happens for any table inheritance chain deeper than two levels.

Not any more! Thanks to this ugly PR, you shall be able to correctly search for Gustavo. (Hint: he's at the toilet.)

### CODE REVIEW

File: plugins/BEdita/Core/src/Model/Behavior/SearchableBehavior.php [hunk 1]
```diff
@@ -159,8 +159,25 @@ function ($word) use ($minLength) {
...
+        );
+        $aliasField = function ($field) use ($tables) {
+            /* @var \Cake\ORM\Table $table */
+            foreach ($tables as $table) {
+                if ($table->hasField($field, false)) {
+                    return $table->aliasField($field);
+                }
+            }
+
+            return $field;
```
Comment (stefanorosanelli) [inline plugins/BEdita/Core/src/Model/Behavior/SearchableBehavior.php]: `codecov` complains that this line is not tested, could we add a simple test case for it?

if problematic we can skip this...

Comment (fquffio) [reply plugins/BEdita/Core/src/Model/Behavior/SearchableBehavior.php]: This was an edge case, so hopefully this line was never hit. However, I have slightly refactored the closure, and now coverage should be 100%. 🎉

**Comment by fquffio** [comment]
This PR also adds a `?q=` shorthand for `?filter[query]=`. This might come in handy in many scenarios. However, it is not fully JSON API compliant, since JSON API _recommends_ to always use `_` or `-` in non-standard query parameters.

I hope this won't keep us awake at night, anyways.

**Review by stefanorosanelli — APPROVED** [review]
